content/ui-patterns/degraded-experiences.mdx

```diff
@@ -0,0 +1,535 @@
+
+<CustomVideoPlayer
+  width="960"
+  loop
+  src="https://github.com/primer/design/assets/2313998/f7837087-a6ff-4572-9ef6-d5dfad158b88"
+/>
+
+### Handling action menus with non-functional items
+
+<img
+  width="960"
+  alt="Three examples: 1. An action menu that works just fine; 2. An action menu with some items disabled; 3. An action menu with all items disabled"
+  src="https://github.com/primer/design/assets/2313998/92ab5cea-74a8-43f0-834c-3ef07ac65bfe"
+/>
+
+Show the menu items, but disable them and show a message explaining why they're disabled.
+
+If an action menu contains menu items that are very closely related (for example: the action menu for adding a file to a repo) and we can detect that none of the menu items are available, then indicate a problem on the button that triggers the action menu. Use an icon in the button to indicate that something is wrong, and (optionally) show a tooltip on hover or focus.
+
+### Handling degraded forms
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  justifyContent="space-between"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 3}}
+>
+<div>
+
+#### Failed submission
+
+If a form's data cannot be saved, don't remove or disable the form fields or the submit button. Instead, inform the user that the form is in a read-only state and cannot be saved. If they still try and submit the form, show a dialog explaining that the form cannot be saved.
+
+</div>
+
+<img
+  width="456"
+  alt="Populated form that cannot be submitted"
+  src="https://github.com/primer/react/assets/2313998/d0494ff2-dea0-4286-b7d6-f6a3d0f6d1db"
+/>
+
+</Box>
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  justifyContent="space-between"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 3}}
+>
+
+<div>
+
+#### No form data is available
+
+If none of the form data cannot even be read, remove the form entirely and replace it with a message explaining why the form is unavailable.
+
+</div>
+
+<img
+  width="456"
+  alt="An error message in replacing a form"
+  src="https://github.com/primer/design/assets/2313998/c15954b0-f64e-4fae-94f5-1c2d474b1e06"
+/>
+</Box>
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  justifyContent="space-between"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 3}}
+>
+
+<div>
+
+#### Form data is partially available
+
+If only some of the form field data cannot be read, disable the affected fields and show an error message below those fields explaining that the data is unavailable. Disabling the fields adds another hint that something is wrong, and the field data is not actually empty.
+
+</div>
+
+<img
+  width="456"
+  alt="A form with some fields disabled. Disabled fields have error messages."
+  src="https://github.com/primer/design/assets/2313998/a869762c-33f5-4cf9-bd95-d76d776f233c"
+/>
```

Comment:
I see that we are using red error color with the warning symbol. I noticed in most other places we use gray and the yellow/amber color to warn users. Should we keep the color consistent instead of showing a red error color?

Reply:
This is just the style we use for `<FormControl.Validation variant="error">`. [Storybook example](https://primer.style/react/storybook/?path=/story/components-textinput--playground&args=disabled:!true;validationChildren:Something+is+wrong).

We're removing "warning" variant from form controls in the next major Primer React release (coming soon!), and I'm not sure if this is a good enough reason to keep it around. cc @joshblack